### ISSUE TITLE
What do the modes __backtest__ and __optimize__ do and how are they different?

### ISSUE THREAD

**Issue by enjoysmath** [EA31337/EA31337#284]
```
// Optional modes (Uncomment to enable).
//#define __backtest__     // For backtest only.
//#define __cli__          // Enables CLI mode.
//#define __debug__        // Enables debugging.
//#define __input__        // Enables user input params.
//#define __optimize__     // Optimization mode.
//#define __profiler__     // Activates profiler.
//#define __property__     // Enables program properties.
//#define __limited__      // Defines safe options.
//#define __release__      // Enables release settings.
//#define __trace__        // Enables tracing.
```

Are all the modes listed in src/include/common/mode.h.  Was wondering if the reason I'm finding it hard to optimize is because I've been optimizing / backtesting in __advanced__ and __lite__ modes...  Please advise.  Should I be using some other modes when I'm picking parameters and running a GA optimization backtest?

**Comment by kenorb** [comment]
There is not much usage at the moment for backtest and optimize modes, it  was more used in v1 version when the code was slower, in v2 the code was refactored to be more optimal, but the general idea is as follow.

Note: Currently below modes are not used so much, as I don't run heavy optimization yet. Tests are usually split to smaller interactions. So below points are for general idea only.

## `__backtest__`

> //#define __backtest__     // For backtest only.

The aim of backtest mode is to improve performance of automated backtesting runs by:

- disabling chart operations such as plotting or printing (still useful for backtest run by user, but not for headless tests);
- disabling log printing or limit to errors only (logs are still useful for tests run by user, but not for automated run when only results are needed);
- disabling lines of code which takes unnecessary time, but are not necessary for the run;
- and so on

## `__optimize__`

> //#define __optimize__     // Optimization mode.

The aim of optimization mode is to improve performance of automated optimization runs by:

- disabling  chart operations such as plotting or printing;
- disabling log printing (platform does something similar, but some code still runs and take time) as optimization passes doesn't give any logs in return either (just some critical errors);
- disabling any non-core functionality such as lines of code which takes unnecessary time, but are not necessary for the run;
- it can enable some extra caching which can make the results quicker, but slightly different, and the overall results still are useful (not used yet);
- it may skip some extra ticks from processing or code for quicker results (not used yet), but having tick filter param, it can be used instead during optimization;
- basically the idea of this mode is to disable everything that is not mandatory which takes time and won't affect the final results at the same time;
- and so on
